sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/TypeUtils.scala

```diff
@@ -62,10 +63,6 @@ object TypeUtils {
   }
 
   def compareBinary(x: Array[Byte], y: Array[Byte]): Int = {
-    for (i <- 0 until x.length; if i < y.length) {
-      val res = x(i).compareTo(y(i))
-      if (res != 0) return res
-    }
-    x.length - y.length
+    UnsignedBytes.lexicographicalComparator().compare(x, y)
```

Comment:
No, not necessary, as `lexicographicalComparator()` used a static compactor instance internally.